src/interface/truncation.jl

```diff
@@ -42,85 +42,102 @@ Truncation strategy that does nothing, and keeps all the values.
+struct TruncationByFilter{F} <: TruncationStrategy
     filter::F
 end
 
 """
-    trunctol(val::Real; by=abs)
+    truncfilter(filter)
 
-Truncation strategy to discard the values that are smaller than `val` according to `by`.
+Truncation strategy to keep the values for which `filter` returns true.
 """
-trunctol(val::Real; by=abs) = TruncationKeepFiltered(≥(val) ∘ by)
+truncfilter(f) = TruncationByFilter(f)
 
 """
-    truncabove(val::Real; by=abs)
+    TruncationByValue(atol::Real, rtol::Real, p::Real, by, rev::Bool=true)
 
-Truncation strategy to discard the values that are larger than `val` according to `by`.
+Truncation strategy to keep the values that satisfy `by(val) < max(atol, rtol * norm(values, p)`
+if `rev = true`, or discard them when `rev = false`.
+See also [`trunctol`](@ref)
 """
-truncabove(val::Real; by=abs) = TruncationKeepFiltered(≤(val) ∘ by)
-
-struct TruncationKeepAbove{T<:Real,P<:Real,F} <: TruncationStrategy
+struct TruncationByValue{T<:Real,P<:Real,F} <: TruncationStrategy
     atol::T
     rtol::T
     p::P
     by::F
+    rev::Bool
 end
-function TruncationKeepAbove(; atol::Real, rtol::Real, p::Real=2, by=abs)
-    return TruncationKeepAbove(atol, rtol, p, by)
+function TruncationByValue(atol::Real, rtol::Real, p::Real=2, by=abs, rev::Bool=true)
+    return TruncationByValue(promote(atol, rtol)..., p, by, rev)
 end
-function TruncationKeepAbove(atol::Real, rtol::Real, p::Real=2, by=abs)
-    return TruncationKeepAbove(promote(atol, rtol)..., p, by)
+
+"""
+    trunctol(; atol::Real=0, rtol::Real=0, p::Real=2, by=abs, )
+
+Truncation strategy to keep the values that satisfy `by(val) < max(atol, rtol * norm(values, p)`
+if `rev = true`, or discard them when `rev = false`.
+"""
+function trunctol(; atol::Real=0, rtol::Real=0, p::Real=2, by=abs, rev::Bool=true)
+    return TruncationByValue(atol, rtol, p, by, rev)
 end
 
 """
-    TruncationKeepBelow(; atol::Real, rtol::Real, p=2, by=abs)
+    TruncationByError(; atol::Real, rtol::Real, p::Real)
 
-Truncation strategy to discard the values that are smaller than the norm of the values.
+Truncation strategy to discard values until the error caused by the discarded values exceeds some tolerances.
+See also [`truncerror`](@ref).
 """
-struct TruncationKeepBelow{T<:Real,P<:Real,F} <: TruncationStrategy
+struct TruncationByError{T<:Real,P<:Real} <: TruncationStrategy
     atol::T
     rtol::T
     p::P
-    by::F
 end
-function TruncationKeepBelow(; atol::Real, rtol::Real, p::Real=2, by=abs)
-    return TruncationKeepBelow(atol, rtol, p, by)
+function TruncationError(atol::Real, rtol::Real, p::Real=2)
+    return TruncationError(promote(atol, rtol)..., p)
 end
-function TruncationKeepBelow(atol::Real, rtol::Real, p::Real=2, by=abs)
-    return TruncationKeepBelow(promote(atol, rtol)..., p, by)
+
+"""
+    truncerror(; atol::Real=0, rtol::Real=0, p::Real=2)
+
+Create a truncation strategy for truncating such that the error in the factorization
```

Comment:
"Create a" only appears in this docstring, whereas the others simply start with "Truncation strategy for ..."